### PR TITLE
Saito Gaming Protocol and Library - Milestone 1 Delivery

### PR DESCRIPTION
**Pull request by arpee**
# Milestone Delivery Checklist

- [*] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [*] The [invoice form :pencil:](https://forms.gle/vgQEZzvvFUjCyBiW6) has been filled out for this milestone. 
- [*] This pull request is being made by the same account as the accepted application. 
- [*] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application. 
- [*] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/General-Grants-Program/blob/master/grants/milestone-deliverables-guidelines.md).

### CODE REVIEW

**Comment by alxs** [comment]
Thank you for the delivery @arpee. We will look into it as soon as possible.

**Comment by semuelle** [comment]
Hi Richard,

- Is there a reason the `tests` folder referenced in the `package.json` is not in the repo? It's not specifically mentioned in your deliverables, but we usually ask for at least *some* test coverage. Manual testing can be quite labor-intensive.
- How do I enable and connect to Westend when running Saito locally? The frontend tells me I can't add modules without the app store being installed. 
- We have a pretty clear [policy](https://github.com/w3f/Open-Grants-Program/pull/73#issuecomment-721714118) against using the W3F name and grants logo before at least the first milestone is approved. Could you please at least remove the link from the header on your website until this milestone is approved?

**Comment by arpee** [comment]
Hi Samuelle - on the use of the name - I have removed this.

It it is late in the day here I will respond in detail tomorrow.

**Comment by semuelle** [comment]
Hi Richard, I also just noticed that the repo doesn't have a license. Could you add one of the licenses suggested in the [application guidelines](https://github.com/w3f/Open-Grants-Program/#guidelines)?

**Comment by arpee** [comment]
Apache 2 License Added to Repository.

**Comment by arpee** [comment]
@semuelle @clayrab from our team has clarified the instructions in the repo around your second bullet point.

Will follow up with more on the first point on tests.

**Comment by arpee** [comment]
@samuelle We have updated the repo to include tests for the delivered software and infrastructure in isolation.

There is now a matching ```Testing``` section in ```README.md``` with instructions.

The tests run by default against the Westen, Kusama and Polkadot nodes deployed by Saito.
Balance transfer tests are commented out by default for non-test-networks for obvious reasons.

TLDR:

```
npm install
npm run test
```

Let me know if you have any questions.

**Comment by semuelle** [comment]
Just FYI, it looks like your Westend endpoint is down. The tests are failing and saito.io shows `DOT`, `KSM` and `Loading...`.

**Comment by semuelle** [comment]
Thanks for following up on the tests and the setup instructions. I just pushed my [evaluation notes](https://github.com/w3f/Grant-Milestone-Delivery/blob/0e9249f5624d53245f6a586871c2c832bd405b37/evaluations/saito_1_semuelle.md) and your milestone is hereby accepted. Congratulations!

Looking forward to your next delivery. I will forward your invoice for processing.

**Comment by arpee** [comment]
> Just FYI, it looks like your Westend endpoint is down. The tests are failing and saito.io shows `DOT`, `KSM` and `Loading...`.

Thanks for that @semuelle - turned out to be ram on the instance - increased ram and updated the binary.

Up and running now.

**Comment by RouvenP** [comment]
hi @arpee we sent a test transaction last Friday. Could you confirm if received?

**Comment by arpee** [comment]
Hello RouvenP,

I can confirm receipt of 0.001 BTC
to bc1qvx6wzyxsjw5yza5f7qj4mprzj4hl49q9js0d46 in TX:
1569b6f29c5d90101d2063722240fde6a3379b30c78992777294a6deafef3e3a.

Thank you,

Richard @arpee

**Comment by RouvenP** [comment]
thanks for confirming @arpee. we just sent the payment.

**Comment by arpee** [comment]
@RouvenP - full payment received. thanks.
